kyaml/yaml/rnode.go

```diff
@@ -12,6 +12,7 @@ import (
 	"strconv"
 	"strings"
 
+	"sigs.k8s.io/kustomize/api/internal/utils"
```

Comment:
We cannot have kyaml import api, because api imports kyaml. We have to move the relevant functions into kyaml.

Reply:
Additionally, things under an `internal` folder are not available outside of the module, so we wouldn't be able to use anything in `api/internal` in the kyaml module anyways.

Reply:
@natasha41575 
Moved the code in `pathsplitter.go` and `pathsplitter_test.go` to kyaml/utils. Still the 3 tests are failing, unable to find out the cause of the errors by the logs.

Could you please inform if you have an idea.

Reply:
Moving the code involves deleting it from the old place as well, after copying to the new place. Wherever it is currently being used will have to import it from kyaml.

Reply:
@natasha41575 
Deleted the files `kustomize/api/internal/utils/pathsplitter.go` and `kustomize/api/internal/utils/pathsplitter_test.go`. Also changed import statements as needed. 

Still few tests are failing, what could be the reason behind ?

Reply:
It looks like some tests related to vars are failing. Off the top of my had, I don't know why. Have you found anything while investigating the test failures?